config/kibana.yml

```diff
@@ -1,8 +1,11 @@
 # Kibana is served by a back end server. This setting specifies the port to use.
 # server.port: 5601
 
-# This setting specifies the IP address of the back end server.
-# server.host: "0.0.0.0"
+# Specifies the host name to which the Kibana server will bind.
+# The default is 'localhost', which usually means remote machines will not be able to connect.
+# To allow connections from remote users, set this parameter to IP or the DNS-name of the machine that is running Kibana.
+# For example: if remote users need to connect to Kibana via http://foobar:5601, set this parameter to 'foobar'.
+# server.host: "localhost"
```

Comment:
> that example works in the local case, but connections won't work from remote hosts.

Sure, if I want to expose Kibana the world, I would enter a publicly accessible IP. In that case my domain name would also get resolved by a DNS server, not /etc/hosts.

> You can use IP (the public IP) for this setting, it's just not a setting I would personally encourage because IP assignment is too low level to bother with IMO.

I disagree, that's exactly the right level to think at. Kibana is deployed on a machine, and Kibana must listen on some address that machine has. This setting has nothing to do with the network external to the machine Kibana is running on (as far as I can tell), it's all about what local address Kibana should listen on. I think what you're really advocating for are some useful aliases, like [elasticsearch provides](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-network.html#network-interface-values). I'd totally support that.